### ISSUE TITLE
[RFC] PD stage 0: transfer foundation (JAX transfer wrapper and connection ABC)

### ISSUE THREAD

**Issue by cjx0709** [sgl-project/sglang-jax#1199]
## Summary

Introduce the lowest-level transfer building blocks for Prefill-Decode Disaggregation (PD) on sgl-jax:

- a process-level wrapper around `jax.experimental.transfer`
- a connection ABC (`KVManager` / `KVSender` / `KVReceiver` / `KVPoll`)
- a `TransferBackend` ABC (`register_pull` / `pull` / `release` per uuid) with a concrete `JaxTransferBackend`
- a byte-correct cross-pod round-trip test for the transfer path itself

This issue is intentionally scoped to the transfer layer only. It does not wire anything into the scheduler, bootstrap, router, or OpenAI entrypoints.

## Position in the roadmap

PD is planned in five increments:

1. Transfer foundation *(this issue)*
2. Host pool + side channel + transport core extraction + multi-entry payload
3. Scheduler + end-to-end
4. Multi-host + routing
5. Production hardening

This issue defines the first increment and establishes the core transfer contracts that all later stages depend on.

## Proposed structure changes

Add or update the following modules:

- `python/sgl_jax/srt/disaggregation/jax_transfer_wrapper.py`
- `python/sgl_jax/srt/disaggregation/base/kv_manager.py`
- `python/sgl_jax/srt/disaggregation/jax_transfer/conn.py`
- `python/sgl_jax/srt/disaggregation/transport/backend.py` — `TransferBackend` ABC + `JaxTransferBackend`
- `python/sgl_jax/srt/disaggregation/transport/core.py` — `RequestTransportCore` (extracted in Stage 1)
- `python/sgl_jax/test/disaggregation/test_jax_transfer_wrapper.py`
- `python/sgl_jax/test/disaggregation/test_kv_manager_state.py`
- `python/sgl_jax/test/disaggregation/test_transfer_backend.py`
- `python/sgl_jax/test/disaggregation/test_byte_roundtrip.py`

### 1. `JaxTransferWrapper`

Introduce a process-level wrapper over `jax.experimental.transfer` with the following responsibilities:

- enforce a single transfer server per process
- make `start()` idempotent
- require explicit sharding on `pull()` input specs
- retain registered buffers until explicit release
- support multi-peer pulls via an explicit `remote_addr`

Target interface:

```python
class JaxTransferWrapper:
    def __init__(self, host_ip: str, port: int, channel_number: int = 1): ...
    def start(self) -> None: ...
    def register_pull(self, uuid: str, data: jax.Array) -> None: ...
    def pull(self, uuid: str, spec: jax.ShapeDtypeStruct, remote_addr: str) -> jax.Array: ...
    def release(self, uuid: str) -> None: ...
```

### 2. Connection ABC

Introduce a backend-agnostic transfer contract in `base/kv_manager.py`:

- `KVPoll`
- `KVManager`
- `KVSender`
- `KVReceiver`

The transfer state machine should allow:

```text
BOOTSTRAPPING -> WAITING_FOR_INPUT -> TRANSFERRING -> SUCCESS
BOOTSTRAPPING -> WAITING_FOR_INPUT -> TRANSFERRING -> FAILED
```

Illegal transitions should raise immediately.

### 3. `TransferBackend` ABC

Introduce a leaf-level tensor transfer abstraction in `transport/backend.py`:

```python
class TransferBackend(ABC):
    def register_pull(self, uuid: str, data: jax.Array) -> None: ...
    def pull(self, uuid: str, spec: jax.ShapeDtypeStruct, *, remote_addr: str) -> jax.Array: ...
    def release(self, uuid: str) -> None: ...
```

`JaxTransferBackend` is the concrete implementation that delegates to `JaxTransferWrapper`. The backend operates at per-uuid granularity and is payload-agnostic — it does not know whether the data is KV cache, embeddings, or anything else.

### 4. Initial concrete backend

Introduce a first `JaxTransferKVManager` backend that exercises the contract end-to-end:

- sender registers a pullable payload
- receiver pulls by `(uuid, spec, remote_addr)`
- state transitions are externally observable through `poll()`

This stage is allowed to remain transfer-local; later stages will replace any synchronous placeholder behavior with side-channel-driven lifecycle completion.

### Unit tests (CPU)

- `test_jax_transfer_wrapper.py`
  - `start()` is idempotent
  - `pull()` rejects specs without sharding
  - wrapper retains and releases registered buffers correctly
- `test_kv_manager_state.py`
  - legal transitions succeed
  - illegal transitions raise
- `test_transfer_backend.py`
  - ABC cannot be instantiated
  - `JaxTransferBackend` delegates correctly to wrapper
  - isinstance check

### Cross-pod validation

- `test_byte_roundtrip.py`
  - validates byte equality for representative dtypes and payload sizes
  - demonstrates that the transfer path itself is correct before any scheduler integration

## Acceptance criteria

- wrapper lifecycle is covered by unit tests
- transfer state transitions are validated by tests
- `TransferBackend` ABC exists with a concrete `JaxTransferBackend`
- a concrete backend exists and follows the ABC contract
- byte-correct transfer validation exists independently of scheduler/bootstrap/router work
